src/server/sched_func.c

```diff
@@ -323,31 +307,35 @@ action_sched_user(attribute *pattr, void *pobj, int actmode)
  *
  * @param[in]	psched	-	Scheduler
  */
-void
-set_sched_default(pbs_sched* psched) {
+void set_sched_default(pbs_sched* psched) {
 	if (!psched)
 		return;
-	if ((psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags & ATR_VFLAG_SET)
-			== 0) {
-		/* unset, reset to default of 20 minutes */
+	if ((psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags
+			& ATR_VFLAG_SET) == 0) {
 		psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_val.at_long =
-				PBS_SCHED_CYCLE_LEN_DEFAULT;
-		psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags = ATR_VFLAG_DEFLT
-				| ATR_VFLAG_SET | ATR_VFLAG_MODCACHE;
+		PBS_SCHED_CYCLE_LEN_DEFAULT;
+		psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags =
+		ATR_VFLAG_DEFLT | ATR_VFLAG_SET | ATR_VFLAG_MODCACHE;
 	}
 	if ((psched->sch_attr[(int) SCHED_ATR_sched_state].at_flags & ATR_VFLAG_SET)
-			==0) {
-		psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str =  malloc(SC_STATUS_LEN + 1);
-		if ( psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str == '\0' )
+			== 0) {
+		psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str = malloc(
+		SC_STATUS_LEN + 1);
+		if (psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str == '\0')
```

Comment:
You're comparing a string to a character.  I suspect you're trying to compare to NULL.